secure/key-manager/src/tests.rs

```diff
@@ -251,58 +208,296 @@ impl LibraInterface for TestLibraInterface {
     }
 }
 
+impl LibraInterfaceWrapper for MockLibraInterface {
+    fn retrieve_discovery_set(&self) -> Result<DiscoverySet, Error> {
+        let account = account_config::discovery_set_address();
+        let account_state = self.retrieve_account_state(account)?;
+        Ok(account_state
+            .get_discovery_set_resource()?
+            .ok_or_else(|| Error::DataDoesNotExist("DiscoverySetResource".into()))?
+            .discovery_set()
+            .clone())
+    }
+
+    fn retrieve_libra_block_resource(&self) -> Result<LibraBlockResource, Error> {
+        let account = account_config::association_address();
+        let account_state = self.retrieve_account_state(account)?;
+        account_state
+            .get_libra_block_resource()?
+            .ok_or_else(|| Error::DataDoesNotExist("BlockMetadata".into()))
+    }
+
+    fn take_all_transactions(&self) -> Vec<Transaction> {
+        self.queued_transactions.borrow_mut().drain(0..).collect()
+    }
+}
+
+#[derive(Clone)]
+/// A JsonRpcLibraInterface wrapper implementation that stores a list of transactions submitted via
+/// the API and exposes additional methods for testing (by implementing the LibraInterfaceWrapper
+/// trait). This is required to better test the JsonRpcLibraInterface implementation.
+struct JsonRpcLibraWrapper {
+    submitted_transactions: Arc<RefCell<Vec<Transaction>>>,
```

Comment:
did you explore how complicated it would be to actually leverage a mempool instead? -- i.e. a fully running or mostly running node?

Reply:
Not yet.. I felt like this is something I should do in a future PR (seeing as though I'm not 100% sure how much would need to change to support this and this PR is already pretty big as it currently stands!).